.pre-commit-config.yaml

```diff
@@ -0,0 +1,52 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.2.0
+    hooks:
+      - id: check-yaml
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+  - repo: https://github.com/python/black
+    rev: 22.3.0
+    hooks:
+      - id: black
+  - repo: https://github.com/PyCQA/flake8
+    rev: 4.0.1
+    hooks:
+      - id: flake8
+        args:
+          - "--count"
+          - "--show-source"
+          - "--statistics"
+  - repo: https://github.com/PyCQA/isort
+    rev: 5.10.1
+    hooks:
+      - id: isort
+  - repo: https://github.com/PyCQA/pydocstyle
+    rev: 5.1.1
+    hooks:
+      - id: pydocstyle
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v0.982
+    hooks:
+      - id: mypy
+  - repo: local
+    hooks:
+      - id: doctest
+        name: Run doctest
+        entry: python -m doctest
+        language: system
+        files: ".+.rst"
+      - id: __init__ + setup.py
+        name: Check __init__ matches setup.py
+        entry: python crosshair/tools/check_init_and_setup_coincide.py
+        language: system
+      - id: help in doc
+        name: Check help in doc
+        entry: python crosshair/tools/check_help_in_doc.py --overwrite
+        language: system
+        pass_filenames: false
+      - id: pytest
```

Comment:
I'm worried that the full test suite may be too much to run per-commit. Do you have a suggestion for what's appropriate here? (as a follow-up, I can work on some sort of "smoke test" phase)

Perhaps we put this in the "push" or "manual" stage?: https://pre-commit.com/#confining-hooks-to-run-at-certain-stages

We DO want all the tests to run in the github action, of course.

Reply:
That's a fair concern. I didn't actually know about the `stages` property, but that's a good idea!  I can add the testing to only run on either pushes or manual, if you have a preference.  Let me know!

`pre-commit run --all-files` should still run all the hooks, if I understand correctly, so I think that would run even ones with stages.

Reply:
Great. Let's do "manual" for now; we can change if needed later.